internal/io/http/rest_sink.go

```diff
@@ -0,0 +1,121 @@
+// Copyright 2024 EMQ Technologies Co., Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package http
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/pingcap/failpoint"
+
+	"github.com/lf-edge/ekuiper/contract/v2/api"
+	"github.com/lf-edge/ekuiper/v2/internal/pkg/httpx"
+	"github.com/lf-edge/ekuiper/v2/pkg/errorx"
+)
+
+type RestSink struct {
+	*ClientConf
+}
+
+func (r *RestSink) Provision(ctx api.StreamContext, configs map[string]any) error {
+	r.ClientConf = &ClientConf{}
+	return r.InitConf("", configs)
+}
+
+func (r *RestSink) Close(ctx api.StreamContext) error {
+	return nil
+}
+
+func (r *RestSink) Connect(ctx api.StreamContext) error {
+	return nil
+}
+
+func (r *RestSink) Collect(ctx api.StreamContext, item api.MessageTuple) error {
+	return r.collect(ctx, item.ToMap())
+}
+
+func (r *RestSink) CollectList(ctx api.StreamContext, items api.MessageTupleList) error {
+	var err error
+	items.RangeOfTuples(func(_ int, tuple api.MessageTuple) bool {
+		err = r.collect(ctx, tuple.ToMap())
+		if err != nil {
+			return false
+		}
+		return true
+	})
+	return err
+}
+
+func (r *RestSink) collect(ctx api.StreamContext, data map[string]any) error {
+	logger := ctx.GetLogger()
+	headers, err := r.parseHeaders(ctx, data)
+	if err != nil {
+		return err
+	}
+	bodyType, err := ctx.ParseTemplate(r.config.BodyType, data)
```

Comment:
Template is now parsing in transform op and pass to sink as "dynamic property". Check mqtt sink line 91about how to use that.